pkg/controller/ps/upgrade.go

```diff
@@ -106,13 +106,46 @@ func (r *PerconaServerMySQLReconciler) smartUpdate(ctx context.Context, sts *app
 
 	log.Info("apply changes to primary pod", "pod", primPod.Name)
 
-	if primPod.ObjectMeta.Labels[controllerRevisionHash] == sts.Status.UpdateRevision {
-		log.Info("primary pod updated updated", "primPod name", primPod.Name)
-		log.Info("smart update finished")
+	if primPod.ObjectMeta.Labels[controllerRevisionHash] != sts.Status.UpdateRevision {
+		log.Info("primary pod was deleted", "pod", primPod.Name)
+		err = deletePod(ctx, r.Client, primPod, currentSet)
+
+		if err != nil {
+			log.Info("primary pod deletion error", "pod", primPod.Name)
+			return err
+		}
+	}
+
+	retriable := func(err error) bool {
+		return err != nil
+	}
+
+	retry := k8sretry.DefaultRetry
+	retry.Duration = 5 * time.Second
+	retry.Steps = 10
```

Comment:
please don't mutate this, you can create a custom `wait.Backoff` object

Reply:
should we update it for deletePod method too?